### PR TITLE
Refresh effect time instead of losing item

### PR DESCRIPTION
**Pull request by agustinesco**
## Motivation

Currently whenever you use an item and you already have that item effect applied you lose that item effect, that's wrong

The correct behavior should be to refresh the item effect duration to his original duration, notice that we don't add up durations, take the next scenario as example:
1. You use an item that lasts 10 seconds
2. After 5 seconds you use that item again
3. The remaining duration of the effect should be 10 again 

## Summary of changes

Change effect application behavior when an item has the `one_time_application` flag in true

## How to test it?

Start a match and use an item twice, the total duration should depend in the second item usage

## Checklist
- [x] Tested the changes locally.
- [x] Reviewed the changes on GitHub, line by line.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by tvillegas98 — COMMENTED** [review]
Seems to work very well! I've a question, why don't we just update the expires_at of the effect instead of creating a new one? just wondering

**Comment by agustinesco** [comment]
I couldn't think in a better updating function in that time 😅  i just did what you suggested and updates the effect instead of deleting it!

**Review by tvillegas98 — DISMISSED** [review]
Great work @agustinesco !

File: apps/arena/lib/arena/game/effect.ex [hunk 1]
```diff
@@ -291,4 +305,15 @@ defmodule Arena.Game.Effect do
...
+  defp refresh_effect_expires_at(effect, start_time) do
+    new_expires_at =
+      if effect.expires_at do
+        start_time + effect.duration_ms
+      else
+        nil
+      end
+
+    Map.put(effect, :expires_at, new_expires_at)
+  end
```
Comment (Nico-Sanchez) [inline apps/arena/lib/arena/game/effect.ex:318]: This function is called inside an if. Why can't you just use `and`?

File: apps/arena/lib/arena/game/effect.ex [hunk 2]
```diff
@@ -26,16 +26,30 @@ defmodule Arena.Game.Effect do
...
+            else
+              effect
+            end
+          end)
+
+        put_in(
+          entity,
+          [:aditional_info, :effects],
+          updated_effects
+        )
```
Comment (Nico-Sanchez) [inline apps/arena/lib/arena/game/effect.ex:51]: If you use `and` this whole block could be the private function.

**Review by Nico-Sanchez — CHANGES_REQUESTED** [review]
Currently the item effects stack. Is that the correct behavior?
For example, you have a giant item effect then you use a purple mirror, they stack up.

**Comment by agustinesco** [comment]
> For example, you have a giant item effect then you use a purple mirror, they stack up.

I think that's the correct behavior indeed, we can double check with the team

File: apps/arena/lib/arena/game/effect.ex (hunk 2 above)
Comment (agustinesco) [reply apps/arena/lib/arena/game/effect.ex]: yeah i think both if path can be replaced with functions, in fact i just discovered a bug, we were always increasing the game_state.last_id even when we were refreshing the expire_time. I just created two private functions for each path

File: apps/arena/lib/arena/game/effect.ex (hunk 1 above)
Comment (agustinesco) [reply apps/arena/lib/arena/game/effect.ex]: Completely agreed

**Review by Nico-Sanchez — APPROVED** [review]

**Review by AminArria — APPROVED** [review]
LGTM

**Pull request merged**